modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java

```diff
@@ -143,15 +135,39 @@ private void _assertPagination(int start, int end, int expectedSize)
 		searchContext.setEnd(end);
 		searchContext.setStart(start);
 
-		Hits hits = indexSearcher.search(searchContext, getDefaultQuery());
+		IdempotentRetryAssert.retryAssert(
```

Comment:
Unfortunately we can't avoid IdempotentRetryAssert here. I know you don't like it, but this is not a Portal integration test - there's no `@Sync` to use, and it's uniform with dozens of other `portal-search-elasticsearch` unit tests using it already. If you want, we can discuss in-depth approaches to remove IdempotentRetryAssert from unit tests, as a task for a separate ticket - but this bug is orthogonal to that, and I'd rather not block fixing it (it affects the entire Search team) unless you already have a solution ready to apply.